From fec001a2c83750c35f47e6e621f3acb22e459c7a Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Wed, 11 Sep 2013 12:25:09 -0700
Subject: [PATCH 79/89] tests: Restrict gpt-header-munge to little-endian
 systems

gpt-header-munge uses perl to manipulate the gpt header for the test. It
only works on 64 bit little-endian systems so restrict it to x86_64
only.

* tests/t0210-gpt-resized-partition-entry-array.sh: Skip if not x86_64
* tests/t0211-gpt-rewrite-header.sh: Skip if not x86_64
---
 tests/t0210-gpt-resized-partition-entry-array.sh | 5 +++++
 tests/t0211-gpt-rewrite-header.sh                | 5 +++++
 2 files changed, 10 insertions(+)

diff --git a/tests/t0210-gpt-resized-partition-entry-array.sh b/tests/t0210-gpt-resized-partition-entry-array.sh
index 8854018..512f342 100755
--- a/tests/t0210-gpt-resized-partition-entry-array.sh
+++ b/tests/t0210-gpt-resized-partition-entry-array.sh
@@ -19,6 +19,11 @@
 . "${srcdir=.}/init.sh"; path_prepend_ ../parted $srcdir
 require_perl_digest_crc_
 
+# gpt-header-munge will fail on big-endian systems
+if test $(uname -m) != x86_64; then
+  skip_ 'this test only works on little-endian systems'
+fi
+
 ss=$sector_size_
 
 N=2M
diff --git a/tests/t0211-gpt-rewrite-header.sh b/tests/t0211-gpt-rewrite-header.sh
index 58625a2..a87e753 100644
--- a/tests/t0211-gpt-rewrite-header.sh
+++ b/tests/t0211-gpt-rewrite-header.sh
@@ -23,6 +23,11 @@
 . "${srcdir=.}/init.sh"; path_prepend_ ../parted $srcdir
 require_perl_digest_crc_
 
+# gpt-header-munge will fail on big-endian systems
+if test $(uname -m) != x86_64; then
+  skip_ 'this test only works on little-endian systems'
+fi
+
 ss=$sector_size_
 
 ns=100         # Initial number of sectors.
-- 
1.8.5.3

